source/binary.cpp

```diff
@@ -377,7 +377,9 @@ spv_result_t Parser::parseInstruction() {
                         << " words, but found " << _.word_index - inst_offset
                         << " words instead.";
   }
-  assert(inst_word_count == words.size());
+  assert(!_.requires_endian_conversion ||
+         (inst_word_count == endian_converted_words.size()));
+  assert(_.requires_endian_conversion || (endian_converted_words.size() == 1));
```

Comment:
But I should just put in a comment that matches what I said in the merge comment!